File: tests/fixtures/skeleton/sample.cs
// Sample C# fixture for Skeletonizer coverage tests.
using System;
using System.IO;

namespace SkeletonSample
{
    public interface IUserRepository
    {
        string GetUser(string id);
    }

    public class UserService : IUserRepository
    {
        private readonly string _dbPath;

        public UserService(string dbPath)
        {
            _dbPath = dbPath;
        }

        public string GetUser(string id)
        {
            var raw = File.ReadAllText(_dbPath);
            // BODY_SENTINEL_DO_NOT_LEAK
            return raw;
        }

        public static string FormatUser(string name, string email)
        {
            // BODY_SENTINEL_DO_NOT_LEAK
            return name + " <" + email + ">";
        }
    }
}

File: tests/fixtures/skeleton/sample.php
<?php
// Sample PHP fixture for Skeletonizer coverage tests.
namespace SkeletonSample;

use Exception;

class UserService
{
    private string $dbPath;

    public function __construct(string $dbPath)
    {
        $this->dbPath = $dbPath;
    }

    public function getUser(string $id): ?array
    {
        $raw = file_get_contents($this->dbPath);
        // BODY_SENTINEL_DO_NOT_LEAK
        $data = json_decode($raw, true);
        return $data[$id] ?? null;
    }
}

function formatUser(array $user): string
{
    // BODY_SENTINEL_DO_NOT_LEAK
    return $user['name'] . ' <' . $user['email'] . '>';
}
